lib/mqtt/mqtt.cc

```diff
@@ -629,17 +625,16 @@ SObj mqtt_connect(Timeout                    *t,
 		return nullptr;
 	}
 
-	// Since we requested a clean session, sessionPresent should be false
-	Debug::Assert(sessionPresent == false,
+	Debug::Assert(sessionPresent == !newSession,
 	              "Successfully connected, but a previous session was "
-	              "unexpectedly present.");
+	              "unexpectedly present or absent.");
```

Comment:
This is confusing, maybe make it conditional on new session to pass present or absent as a for,at argument?